WordPress/src/main/java/org/wordpress/android/ui/people/WPEditTextWithChipsOutlined.kt

```diff
@@ -0,0 +1,729 @@
+                                hint.visibility = View.VISIBLE
+                            }
+
+                            override fun onAnimationEnd(animation: Animator?) {
+                            }
+
+                            override fun onAnimationCancel(animation: Animator?) {
+                            }
+
+                            override fun onAnimationRepeat(animation: Animator?) {
+                            }
+                        }).start()
+            }
+            HelperTextState.LABEL_VISIBLE -> {
+                hint.animate().cancel()
+
+                if (hint.translationY != 0f) {
+                    hint.translationY = 0f
+                    hint.scaleX = 1f
+                    hint.scaleX = 1f
+                }
+
+                hint.animate()
+                        .translationY(label.y - hint.y)
+                        .scaleX(label.width.toFloat() / hint.width)
+                        .scaleY(label.height.toFloat() / hint.height)
+                        .setDuration(duration)
+                        .setListener(object : Animator.AnimatorListener {
+                            override fun onAnimationStart(animation: Animator?) {
+                                setLabelColor(label, colorSurface, outlineColorAlphaFocused)
+                                label.visibility = View.VISIBLE
+                                hint.visibility = View.VISIBLE
+                            }
+
+                            override fun onAnimationEnd(animation: Animator?) {
+                                hint.visibility = View.INVISIBLE
+                                setLabelColor(label, outlineColorFocused, outlineColorAlphaFocused)
+                            }
+
+                            override fun onAnimationCancel(animation: Animator?) {
+                            }
+
+                            override fun onAnimationRepeat(animation: Animator?) {
+                            }
+                        }).start()
+            }
+        }
+    }
+
+    private fun setOutlineStroke(width: Int, color: Int, alpha: Float) {
+        outlineDrawable.apply {
+            this.setStroke(width.toFloat(), color)
+            this.alpha = (alpha * 255).coerceAtMost(255f).toInt()
+            this.fillColor = ColorStateList.valueOf(Color.TRANSPARENT)
+        }
+    }
+
+    private fun setLabelColor(textView: MaterialTextView, color: Int, alpha: Float) {
+        val colorWithAlpha = ColorUtils.setAlphaComponent(color, (alpha * 255).coerceAtMost(255f).toInt())
+        textView.setTextColor(colorWithAlpha)
+    }
+
+    private fun hasItemsOrText() = hasItems() || hasText()
+    private fun hasItems() = flexbox.childCount > 1
+    private fun hasText() = (editor.text?.length ?: 0) > 0
+    private fun isEditorFocused() = editor.isFocused
+    private fun hasHint() = hintResourceId != 0
+
+    override fun onSaveInstanceState(): Parcelable? {
```

Comment:
Since we are creating a `SavedState`, I don't think it's necessary to mark return object as nullable.

Reply:
I used auto complete to create the function prototype and was going to say I would prefer to keep it, but checked that kt complains if you return a nullable type there, so changed it 👍